specifyweb/frontend/js_src/lib/components/QueryBuilder/FieldFilter.tsx

```diff
@@ -243,34 +479,111 @@ function SingleField({
 
   readonly enforceLengthLimit: boolean;
 }): JSX.Element {
+  if (parser.type === 'date') {
+    return (
+      <DateQueryInputField
```

Comment:
`DateQueryInputField` and `SpecifyUserAutoComplete` should display as read-only when `handleChange` is undefined (when user is in read only mode)